src/proj.h

```diff
@@ -243,58 +258,79 @@ typedef struct projCtx_t PJ_CONTEXT;
 typedef int *PJ_FILE;
 
 
+/* Functionality for handling thread contexts */
+PJ_CONTEXT *proj_context_create (char *setup, ...);
+void proj_context_set (PJ *P, PJ_CONTEXT *ctx);
+void proj_context_inherit (PJ *parent, PJ *child);
+void proj_context_destroy    (PJ_CONTEXT *ctx);
+
+
 /* Manage the transformation definition object PJ */
-PJ  *pj_create (const char *definition);
-PJ  *pj_create_argv (int argc, char **argv);
-void pj_free (PJ *P);
+PJ  *proj_create (PJ_CONTEXT *ctx, const char *definition);
+PJ  *proj_create_argv (PJ_CONTEXT *ctx, int argc, char **argv);
+PJ  *proj_create_pipeline (PJ_CONTEXT *ctx, const char *def_left, const char *def_right);
```

Comment:
I would prefer `def_from` and `def_to` instead. The idea is to have a function that let's a user create a pipeline that converts from one CRS to another CRS. Example:

```
PJ pipe = proj_create_pipeline(ctx, "+init:epsg:25832", "+init=epsg:25833");
```

which can then be used by the transformation functions. For now it will be implemented as a simple pipeline that uses WGS84 as a hub (early binding), but eventually it can be expanded to use a more complicated setup based on the EPSG-database that makes the direct transformation (late binding).